### PR TITLE
Tutorial enhancements

### PR DESCRIPTION
**Pull request by mstahv**
This needs to be landed at the same time when merging the final code in https://github.com/vaadin/tutorial/tree/flow-tutorial-update-2019-01-31

### CODE REVIEW

File: tutorials/getting-started-with-flow/02__listing-entities-in-a-grid/content.adoc [hunk 1]
```diff
@@ -14,17 +14,19 @@ In this part of the tutorial we cover how to display data in a grid in your UI.
 Before getting into actual Vaadin development, we add the necessary domain objects and a dummy backend. In a real-world application, you would most probably have something similar implemented with JPA and EJB, or a Spring-based service.
 
 The first step is to copy three Java classes from GitHub and add them to your project. 
-We use these classes to build the UI. 
+We will use these classes to when building the UI.
```
Comment (ruthstocks) [inline tutorials/getting-started-with-flow/02__listing-entities-in-a-grid/content.adoc]: Using present tense and active voice when possible is recommended for technical docs. I have also used it throughout. I recommend using the original.

File: tutorials/getting-started-with-flow/02__listing-entities-in-a-grid/content.adoc [hunk 2]
```diff
@@ -14,17 +14,19 @@ In this part of the tutorial we cover how to display data in a grid in your UI.
...
+* https://raw.githubusercontent.com/vaadin/tutorial/vaadin10%2B/src/main/java/com/vaadin/starter/skeleton/backend/CustomerStatus.java[`CustomerStatus.java`]: This is a simple enum class
+* https://raw.githubusercontent.com/vaadin/tutorial/vaadin10%2B/src/main/java/com/vaadin/starter/skeleton/backend/Customer.java[`Customer.java`]: This this is the main domain object. It is the basic Java bean that we use in our example.
+* https://raw.githubusercontent.com/vaadin/tutorial/vaadin10%2B/src/main/java/com/vaadin/starter/skeleton/backend/CustomerService.java[`CustomerService.java`]: This is a simple service class via which you can request and modify `Customer` instances. Think of this class as the entry point to the dummy database.
 
+[TIP]
+The fastest way to copy classes using Eclipse is to use your good, old clipboard. Select the text content of the whole class from your browser, choose "Edit > Copy", focus the node representing the [packagename]#com.vaadin.starter.skeleton.my-app# Java package in Eclipse’s Java Resources view and choose "Edit > Paste". Eclipse is smart enough to automatically create a properly named Java file for the class.
```
Comment (ruthstocks) [inline tutorials/getting-started-with-flow/02__listing-entities-in-a-grid/content.adoc]:
```suggestion
The fastest way to copy classes in Eclipse is to use your clipboard. In your browser, select the text content of the class and choose *Edit > Copy*. In *Project Explorer* in Eclipse, right-click `../[packagename]#com.vaadin.starter.skeleton.my-app#` in `Java Resources` and choose *Edit > Paste*. Eclipse is smart enough to automatically create a properly-named Java file for the class.
```

File: tutorials/getting-started-with-flow/04__creating-a-form-using-plain-java/content.adoc [hunk 3]
```diff
@@ -77,7 +75,7 @@ status.setItems(CustomerStatus.values());
 +
 
 
-. In `CustomerForm.java`, complete the form by creating a public API. This allows you to pass a selected item from `MainView.java` to `CustomerForm.java`. 
+. In `CustomerForm.java`, we need a public API for the `MainView` to define which `Customer` to edit. We will also use `Binder` to make two-way databinding between the properties of the `Customer` and the UI fields in the form.
```
Comment (ruthstocks) [inline tutorials/getting-started-with-flow/04__creating-a-form-using-plain-java/content.adoc]:
```suggestion
. In `CustomerForm.java`, we need a public API for the `MainView` to define which `Customer` to edit. We also use `Binder` for two-way data binding between the properties of the `Customer` and the UI fields in the form. 
```

File: tutorials/getting-started-with-flow/04__creating-a-form-using-plain-java/content.adoc [hunk 4]
```diff
@@ -86,9 +84,7 @@ status.setItems(CustomerStatus.values());
 private Binder<Customer> binder = new Binder<>(Customer.class);
 ----
 
-* A `Binder` is an object that connects data to a form that is defined in the declarative format.
-* You can use a binder class to define how the values in a business object should be bound to the fields shown in the UI. 
-The binder takes care of reading values from the business object, and converts user data from the format expected by the field to that expected by the business object. 
+* A `Binder` is an object that synchronize the properties from domain objects to the UI fields. The binder takes care of reading values from the business object, and converts user data from the format expected by the field to that expected by the business object.
```
Comment (ruthstocks) [inline tutorials/getting-started-with-flow/04__creating-a-form-using-plain-java/content.adoc]:
```suggestion
* A `Binder` is an object that synchronizes the properties of domain objects to the UI fields. The binder takes care of reading values from the business object, and converts user data from the format expected by the field to that expected by the business object. 
```

File: tutorials/getting-started-with-flow/04__creating-a-form-using-plain-java/content.adoc [hunk 5]
```diff
@@ -198,7 +202,7 @@ save.addThemeVariants(ButtonVariant.LUMO_PRIMARY);
 
 * This improves the UI by highlighting the more common action. 
 
-. Disable the save and cancel buttons when no customer is selected, by amending the `setCustomer` method to:
+. Let's also improve the user experience by hidind the form when it is not bound to a domain object and focusing the first name text field when a user is selected. Amend the `setCustomer` method to:
```
Comment (ruthstocks) [inline tutorials/getting-started-with-flow/04__creating-a-form-using-plain-java/content.adoc]:
```suggestion
. We also improve the user experience by hiding the form when it is not bound to a domain object, and focus the first name text field when a user is selected. Amend the `setCustomer` method to:
```

File: tutorials/getting-started-with-flow/04__creating-a-form-using-plain-java/content.adoc [hunk 6]
```diff
@@ -207,25 +211,21 @@ public void setCustomer(Customer customer) {
...
         firstName.focus();
     }
 }
 ----
-* Disabling the buttons discourages user interaction when there is no selection. 
-* As an alternative, you could disable the input fields, by calling the `setEnabled()` method on each. 
-* Setting the focus on the `firstName` field improves the user experience.
 
-. Finalize the `setCustomer` setter method, by calling `setCustomer(null)` at the end of the constructor.  
+. To initialize form state without an active customer, call `setCustomer(null)` at the end of the constructor.
```
Comment (ruthstocks) [inline tutorials/getting-started-with-flow/04__creating-a-form-using-plain-java/content.adoc]:
```suggestion
. To initialize the form state without an active customer, call `setCustomer(null)` at the end of the constructor.  
```

**Comment by mstahv** [comment]
Merged manually.

**Pull request closed without merging**